# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/amanraj310305-tech/flyrank_1/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I want to investigate which pages should be reviewed first for refresh, expansion, protection, pruning, or monitoring. I chose this lane because the starter dataset contains observable search, traffic, content-age, and trend signals that can help prioritize pages for limited human review. The goal is not to guarantee that a page will improve after an action, but to build a useful decision-support ranking.

In [29]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
print("Chosen lane: Refresh / Content Opportunity Scoring")
print("Dataset rows:", len(df))

Chosen lane: Refresh / Content Opportunity Scoring
Dataset rows: 30000


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Decision: Which pages should a content reviewer prioritize first?

Action: A reviewer can inspect the highest-priority pages and decide whether each page should be refreshed, expanded, protected, pruned, or monitored.

Cost of a wrong recommendation: A false positive can waste limited review time on a page that does not need attention. A false negative can cause an important page to be missed or reviewed too late. Therefore, the goal is to prioritize useful candidates rather than make guaranteed predictions.

In [30]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
print("Pages available for prioritization:", len(df))
print("Pages currently showing a down trend:", (df["trend_direction"] == "down").sum())

Pages available for prioritization: 30000
Pages currently showing a down trend: 16262


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [31]:
import os

if not os.path.exists("/content/flyrank_1"):
    !git clone https://github.com/amanraj310305-tech/flyrank_1.git

%cd /content/flyrank_1

/content/flyrank_1


In [32]:
!ls data/raw

content_refresh_anonymized.csv


In [33]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Rows:", len(df))
print("Columns:", len(df.columns))

Rows: 30000
Columns: 44


In [34]:
print("Average 90-day impressions:", df["impressions_90d"].mean())
print("Average 90-day sessions:", df["sessions_90d"].mean())
print("Trend distribution:")
print(df["trend_direction"].value_counts())

Average 90-day impressions: 5200.3663
Average 90-day sessions: 37.066633333333336
Trend distribution:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


The starter dataset contains 30,000 content pages. The average page has about 5,200 impressions and 37 sessions over the last 90 days. There are 16,262 pages with a downward trend, which is about 54% of the starter dataset. These numbers make refresh/content opportunity scoring worth investigating because there is a substantial set of pages showing movement or potential weakness, while the impressions and sessions provide measurable signals for prioritizing human review. This does not mean all declining pages need a refresh or that a refresh will improve performance.

In [35]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Total pages:", len(df))
print("Average 90-day impressions:", df["impressions_90d"].mean())
print("Average 90-day sessions:", df["sessions_90d"].mean())

trend_counts = df["trend_direction"].value_counts()
print("\nTrend distribution:")
print(trend_counts)

print("\nDown-trend percentage:",
      round((trend_counts["down"] / len(df)) * 100, 2), "%")

Total pages: 30000
Average 90-day impressions: 5200.3663
Average 90-day sessions: 37.066633333333336

Trend distribution:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Down-trend percentage: 54.21 %


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

What I can claim: I can identify observable patterns in search, traffic, content-age, and trend signals and use them to create a decision-support ranking of pages for human review. The results will be directional and based on measured data.

What I cannot claim: I cannot claim that refreshing a page will definitely increase its traffic, that my ranking proves causality, or that it predicts Google's algorithm. A high-priority page should be treated as a review candidate, not as a guaranteed opportunity.

In [36]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
print("Rows:", len(df))
print("Columns:", len(df.columns))

unsafe_terms = ["url", "query", "title", "domain"]
found = [col for col in df.columns
         if any(term in col.lower() for term in unsafe_terms)]

print("Potential raw sensitive columns found:", found)

Rows: 30000
Columns: 44
Potential raw sensitive columns found: []


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.